src/SDL.cpp

```diff
@@ -0,0 +1,60 @@
+#include "SDL.h"
+
+#define SCREEN_HEIGHT 768
+#define SCREEN_WIDTH 1232
+
+SDL::SDL() {
+    window = NULL;
+    renderer = NULL;
+}
+
+bool SDL::init_SDL()
+{
+    bool success = true;
+
+    if (SDL_Init(SDL_Init(SDL_INIT_VIDEO | SDL_INIT_AUDIO | SDL_INIT_TIMER)) < 0)
+    {
+        printf("SDL could not initialize! SDL Error: %s\n", SDL_GetError());
+        success = false;
+    }
+    else
+    {
+        window = SDL_CreateWindow("Video Application", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, SCREEN_WIDTH, SCREEN_HEIGHT, SDL_WINDOW_SHOWN);
+        if (window == NULL)
+        {
+            printf("Window could not be created! SDL Error: %s\n", SDL_GetError());
+            success = false;
+        }
+        else
+        {
+            renderer = SDL_CreateRenderer(window, -1, SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC);
```

Comment:
I've noticed the -1 doesn't work on some computers. Change this to 0, or, do the -1 first, and if it's renderer == NULL, then try the 0, and if that doesn't work, then fall back to the SDL_RENDERER_SOFTWARE.